### ISSUE TITLE
NN: Training script

### ISSUE THREAD

**Issue by petrucristescu** [petrucristescu/lamc#67]
## Summary
Create `examples/digits/train.ch` — the main training script.

Part of #59. Depends on #65, #66.

## Behavior
1. Load all PGM images from `data/train/`
2. Parse labels from filenames
3. Run SGD for N epochs with mini-batches
4. Print loss each epoch
5. Save trained weights to `weights.json`

## Usage
```bash
./run.sh examples/digits/train.ch
```

**Comment by petrucristescu** [comment]
Implemented in feat commit.
